demo/src/main/frontend/src/component/editCard/EditCard.js

```diff
@@ -0,0 +1,169 @@
+import Navbar from "../navbar/Navbar";
+import { useParams } from "react-router-dom"
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from "react-redux";
+import axios from 'axios';
+
+function EditCard(props){
+
+    const params = useParams() //params.adsId
+    const [data, setData] = useState();
+    const dispatch = useDispatch();
+    const [adsTitle, setadsTitle] = useState();
+    const [adsLocation, setadsLocation] = useState();
+    const [adsDescription, setadsDescription] = useState();
+    const [category, setCategory] = useState(0);
+    const [adsImage, setadsImage] = useState();
+    const [loading, setLoading] = useState(false);
+    const [image, setImage] = useState("")
+
+    let file
+    const state = useSelector((state) => {
+     return {
+       user: state.userReducer,
+       token: state.userReducer.token
+     };
+   });
+ 
+    let id = state.user.user.id
+    let userToken = state.token
+    const config = {
+     headers: {Authorization: `Bearer ${userToken}`}
+    };
+ 
+    
+  
+    useEffect(async () => {
+        let result = await axios(
+          `http://localhost:8081/api/ads/${params.adsId}` 
+        );
```

Comment:
Why let and not const?  also, it should be `axios.get`, not the best way to use it